mlir/lib/Conversion/ConvertToLLVM/ConvertToLLVMPass.cpp

```diff
@@ -61,6 +61,8 @@ class LoadDependentDialectExtension : public DialectExtensionBase {
 /// the injection of conversion patterns.
 class ConvertToLLVMPass
     : public impl::ConvertToLLVMPassBase<ConvertToLLVMPass> {
+  std::shared_ptr<const SmallVector<ConvertToLLVMPatternInterface *>>
+      interfaces;
   std::shared_ptr<const FrozenRewritePatternSet> patterns;
   std::shared_ptr<const ConversionTarget> target;
   std::shared_ptr<const LLVMTypeConverter> typeConverter;
```

Comment:
There should be much more documentation around this, because you're introducing subtle invariants where fields are only used in one of two modes, and uninitialized in the other. This kind of modal behavior isn't great in general.


Can we actually refactor this by encapsulating this into another abstract class, with two implementations and then unifying all the downstream code from there?

Reply:
Done, I created to sub classes `StaticConvertToLLVM` and `DynamicConvertToLLVM` controlled by the  `convert-to-llvm` `dynamic` flag.